### PR TITLE
Add ability to assign a menu when creating a window on Windows

### PR DESCRIPTION
**Pull request by madsmtm**
- [x] Tested on all platforms changed
- [x] Compilation warnings were addressed
- [x] `cargo fmt` has been run on this branch
- [x] `cargo doc` builds successfully
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

Added `WindowBuilderExtWindows::with_menu`.

The primary reason for doing it here instead of just assigning a menu later on with `winapi::um::winuser::SetMenu` is to avoid that `with_inner_size` breaks, since the menu takes up some space that is not taken into account, which means the user would have to use `set_inner_size` again after adding the menu.
This, I'm guessing, could cause a jarring effect for the user (that the window resizes when starting).

Unresolved:
- [x] ~Child windows cannot have a menu, so perhaps this should be represented typewise?~

### CODE REVIEW

File: src/platform_impl/windows/mod.rs [hunk 1]
```diff
@@ -17,7 +17,9 @@ use crate::window::Theme;
 
 #[derive(Clone)]
 pub struct PlatformSpecificWindowBuilderAttributes {
+    // TODO: parent and menu are mutually exclusive; a child window cannot have a menu!
```
Comment (msiglreith) [inline src/platform_impl/windows/mod.rs:20]: This should be added the `with_menu` documentationa and I would say a `warn` during the creation if that case is hit would be enough

**Review by msiglreith — COMMENTED** [review]
Thanks! LGTM

**Comment by madsmtm** [comment]
Done. Should I do a rebase?

It also occurs to me, both `with_parent_window` and `with_menu` are probably unsafe, since you could safely pass just about any pointer to these, and bad things would probably happen then?

Also, `with_win32_menu` is maybe a better name? Though not so important.

**Comment by msiglreith** [comment]
> Done. Should I do a rebase?

Thanks - no, we squash-merge here :D

> It also occurs to me, both with_parent_window and with_menu are probably unsafe, since you could safely pass just about any pointer to these, and bad things would probably happen then?

Agree, but I think that's a different issue right now, I would like to handle this separately. There is also an open issue regarding exposing typedefs like `HWND` or `HMENU` in public API without exporting.

> Also, with_win32_menu is maybe a better name? Though not so important.

The current name looks totally fine to me!

**Review by msiglreith — APPROVED** [review]

**Pull request merged**